assets/js/modules/analytics/components/settings/GA4SettingsControls.js

```diff
@@ -0,0 +1,170 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import { useUpdateEffect } from 'react-use';
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Data from 'googlesitekit-data';
+import { CORE_FORMS } from '../../../../googlesitekit/datastore/forms/constants';
+import {
+	MODULES_ANALYTICS_4,
+	PROPERTY_CREATE,
+	WEBDATASTREAM_CREATE,
+} from '../../../analytics-4/datastore/constants';
+import { FORM_SETUP, MODULES_ANALYTICS } from '../../datastore/constants';
+import { Select, Option } from '../../../../material-components';
+import { GA4ActivateSwitch } from '../common';
+import {
+	PropertySelect,
+	UseSnippetSwitch,
+} from '../../../analytics-4/components/common';
+const { useSelect, useDispatch } = Data;
+
+export default function GA4SettingsControls() {
+	const [ matchedProperty, setMatchedProperty ] = useState();
+	const [ matchedWebDataStream, setMatchedWebDataStream ] = useState();
+
+	const accountID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS ).getAccountID()
+	);
+
+	// This select is needed to check whether the AdminAPI works or not.
+	useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getProperties( accountID )
+	);
+
+	const isAdminAPIWorking = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).isAdminAPIWorking()
+	);
+
+	const enableGA4 = useSelect( ( select ) =>
+		select( CORE_FORMS ).getValue( FORM_SETUP, 'enableGA4' )
+	);
+
+	const propertyID = useSelect( ( select ) =>
+		select( MODULES_ANALYTICS_4 ).getPropertyID()
+	);
+
+	const {
+		matchAccountProperty,
+		matchWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	} = useDispatch( MODULES_ANALYTICS_4 );
+
+	useUpdateEffect( () => {
+		const matchGA4Information = async () => {
+			const matchingProperty = await matchAccountProperty( accountID );
+
+			setMatchedProperty( matchingProperty );
+			if ( matchingProperty?._id ) {
+				const matchingWebDataStream = await matchWebDataStream(
+					matchingProperty._id
+				);
+				setMatchedWebDataStream( matchingWebDataStream );
+			}
+		};
+
+		if ( isAdminAPIWorking ) {
+			matchGA4Information();
+		}
+	}, [
+		accountID,
+		matchAccountProperty,
+		matchWebDataStream,
+		isAdminAPIWorking,
+	] );
+
+	const onActivate = useCallback( () => {
+		setPropertyID( matchedProperty?._id || PROPERTY_CREATE );
+		setWebDataStreamID( matchedWebDataStream?._id || WEBDATASTREAM_CREATE );
+		setMeasurementID( matchedWebDataStream?.measurementId || '' ); // eslint-disable-line sitekit/acronym-case
+	}, [
+		matchedProperty,
+		matchedWebDataStream,
+		setPropertyID,
+		setWebDataStreamID,
+		setMeasurementID,
+	] );
+
+	if ( ! isAdminAPIWorking ) {
+		return null;
+	}
```

Comment:
This check results in a layout shift when the edit screen is loading which would be good to avoid. How about instead of returning `null` here, we conditionally apply a `visibility: hidden` style while `isAdminAPIWorking !== true` ?

I would even be open to rendering the div at 50% opacity + disabled until the API is confirmed to be working to show that something is there rather than empty space, but either of these would be preferable to the shift we have now.

Thoughts @felixarntz ?

Reply:
@aaemnnosttv Good point, however I'd prefer a loading indicator here rather than showing the UI. @eugene-manuilov Could we return a `ProgressBar` here that matches the height of the overall regular GA4 settings controls content?

Regarding the layout shift, it cannot be 100% avoided here: For example, if the GA4 API is not working, we don't want to keep displaying a disabled or blank-space area here, so in that case there would be a layout shift. With that said though, that should be more of the edge-case situation, so we can optimize for the common case. That's why I'm suggesting to display a progress bar with the height of the GA4 settings controls, to avoid the layout shift in case things work as expected.

If we can get this in today, let's do it - otherwise I'm also okay with fixing this in a separate follow-up issue (conscious of time here since this is a big PR and the release is close).

Reply:
@aaemnnosttv @felixarntz I have added the progress bar. Please, take a look at it and let me know if this looks good to you.